### PR TITLE
Dev/vincent.coubard/perf improvement

### PR DESCRIPTION
**Pull request by pan-**
Improve performances of OIS port: 

- Remove DHCP option from LWIP as we are auto assigning an IPv6 address and don't need an IPv4 one.
- Set correct optimisation level to TFM build to match Mbed TLS level of optimisation.

### CODE REVIEW

File: config/openiotsdk/cmake/sdk.cmake [hunk 1]
```diff
@@ -90,7 +90,7 @@ if(TFM_SUPPORT)
     set(TFM_PLATFORM ${OPEN_IOT_SDK_EXAMPLE_COMMON}/tf-m/targets/an552)
     set(TFM_PSA_FIRMWARE_UPDATE ON)
     set(MCUBOOT_IMAGE_VERSION_NS ${TFM_NS_APP_VERSION})
-    set(TFM_CMAKE_ARGS "-DCONFIG_TFM_ENABLE_FP=ON;-DTFM_PROFILE=profile_medium;-DTFM_EXCEPTION_INFO_DUMP=ON;-DCONFIG_TFM_HALT_ON_CORE_PANIC=ON;-DTFM_ISOLATION_LEVEL=1;-DTFM_MBEDCRYPTO_PLATFORM_EXTRA_CONFIG_PATH=${OPEN_IOT_SDK_CONFIG}/mbedtls/mbedtls_config_psa.h")
+    set(TFM_CMAKE_ARGS "-DCONFIG_TFM_ENABLE_FP=ON;-DTFM_PROFILE=profile_medium;-DTFM_EXCEPTION_INFO_DUMP=ON;-DCONFIG_TFM_HALT_ON_CORE_PANIC=ON;-DTFM_ISOLATION_LEVEL=1;-DTFM_MBEDCRYPTO_PLATFORM_EXTRA_CONFIG_PATH=${OPEN_IOT_SDK_CONFIG}/mbedtls/mbedtls_config_psa.h;-DMBEDCRYPTO_BUILD_TYPE=Release;-DCMAKE_BUILD_TYPE=Release")
```
Comment (ATmobica) [inline config/openiotsdk/cmake/sdk.cmake:93]: Can we use the build type from the main build settings?
```suggestion
    set(TFM_CMAKE_ARGS "-DCONFIG_TFM_ENABLE_FP=ON;-DTFM_PROFILE=profile_medium;-DTFM_EXCEPTION_INFO_DUMP=ON;-DCONFIG_TFM_HALT_ON_CORE_PANIC=ON;-DTFM_ISOLATION_LEVEL=1;-DTFM_MBEDCRYPTO_PLATFORM_EXTRA_CONFIG_PATH=${OPEN_IOT_SDK_CONFIG}/mbedtls/mbedtls_config_psa.h;-DMBEDCRYPTO_BUILD_TYPE=${CMAKE_BUILD_TYPE};-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
```

Comment (pan-) [reply config/openiotsdk/cmake/sdk.cmake]: solved

File: config/openiotsdk/mbedtls/mbedtls_config_psa.h [hunk 2]
```diff
@@ -1,3 +1,6 @@
 
 #define MBEDTLS_SHA1_C
 #define PSA_WANT_ALG_SHA_1
+#define MBEDTLS_ECP_WINDOW_SIZE 6 /**< Maximum window size used */
+#undef MBEDTLS_ECP_FIXED_POINT_OPTIM
+#define MBEDTLS_ECP_FIXED_POINT_OPTIM 1 /**< Enable fixed-point speed-up */
```
Comment (ATmobica) [inline config/openiotsdk/mbedtls/mbedtls_config_psa.h:6]: New line at the end of the file.

File: src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py [hunk 3]
```diff
@@ -115,7 +115,7 @@ def test_lock_ctrl(device, controller):
                                      userStatus=DoorLock.Enums.UserStatusEnum.kOccupiedEnabled,
                                      userType=DoorLock.Enums.UserTypeEnum.kUnrestrictedUser,
                                      credentialRule=DoorLock.Enums.CredentialRuleEnum.kSingle),
-                                requestTimeoutMs=1000)
+                                requestTimeoutMs=2000)
```
Comment (ATmobica) [inline src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py:118]: I'm a bit surprised that we need so much time to handle the command

File: config/openiotsdk/mbedtls/mbedtls_config_psa.h (hunk 2 above)
Comment (pan-) [reply config/openiotsdk/mbedtls/mbedtls_config_psa.h]: solved

File: src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py (hunk 3 above)
Comment (pan-) [reply src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py]: I removed the commit

**Review by ATmobica — APPROVED** [review]

**Pull request merged**